### DISCUSSION TITLE
Research and document the exact models of WiFi co-processor boards we can support

### DISCUSSION THREAD

**Discussion by calebbourg** [Jim-Hodapp-Coaching/esp32-wroom-rp#5] [Design]
- Research and document the exact models of WiFi co-processor boards we can support, what are the differences?
- What major version(s) of the Adafruit (and Arduino) ESP32 WiFi co-processor firmware do we want to support?

**Comment by jhodapp** [comment]
In order to ensure our crate remains useful outside of only supporting the [Pimoroni Pico Wireless pack](https://shop.pimoroni.com/products/pico-wireless-pack?variant=32369508581459), we also want to make sure that we understand what it means to support others like the [Adafruit Airlift](https://learn.adafruit.com/adafruit-airlift-breakout) or the [Adafruit HUZZAH32](https://www.adafruit.com/product/3405?gclid=CjwKCAjw9qiTBhBbEiwAp-GE0VO7b2rRvqthVetA8tfpcLGt24dRXEf88CYUfptjX_52KzNxl4qScRoCIIoQAvD_BwE). Some questions to research and answer that will influence the design/layout of our crate:

Do the Pimoroni and Adafruit WiFi co-processor boards both run the exact same firmware on their ESP32-WROOM modules?
If they're different, what are the differences (e.g. do they use the same SPI protocol, if the protocol differs, are they similar enough to abstract under a single HAL)?

I found [one interesting discussion](https://www.mjoldfield.com/atelier/2021/11/pico-wireless.html), [attached it as HTML](https://github.com/Jim-Hodapp-Coaching/esp32-wroom-rp/files/8647938/pimoroni-pico-wifi.html.zip) since the original post was rendering the HTML page as TXT, so if you view this locally on your web browser it'll render as actual HTML.

**Comment by calebbourg** [comment]
We are planning on experimenting with adafruit's [Adafruit AirLift – ESP32 WiFi Co-Processor Breakout Board
Product ID: 42](https://www.adafruit.com/product/4201) as our second supported board.

**Design Decisions:**
- For our initial project we will only support core WIFI functionality.
- We will not support board specific features.

**Resources:**
- [Pimoroni Pico Wireless pack](https://shop.pimoroni.com/products/pico-wireless-pack?variant=32369508581459)
- [Adafruit firmware](https://github.com/adafruit/nina-fw) (forked from arduino)
- [Arduino firmware API docs](https://www.arduino.cc/reference/en/libraries/wifinina/)
- [Espressif ESP32-wroom-32 Datasheet ](https://www.espressif.com/sites/default/files/documentation/esp32-wroom-32e_esp32-wroom-32ue_datasheet_en.pdf)

**Comment by jhodapp** [comment]
I was able to get the Airlift working locally without a single change to our [esp32-pico-wifi PoC project](https://github.com/Jim-Hodapp-Coaching/esp32-pico-wifi). There was some difference in behavior but I believe that's because of two reasons:

- We have a suspected issue where we're trying to communicate with the ESP32 co-processor while it isn't accepting commands
- Different nina-fw versions

Here's an image of the Airlift attached to my second Pico setup next to the Pimoroni.

![IMG_3860](https://user-images.githubusercontent.com/3219120/169721165-1394f85e-7406-45d0-bd28-7d78cbe1ea95.png)
